### ISSUE TITLE
TalkBack outline with a FlutterFragment is buggy

### ISSUE THREAD

**Issue by dagonco** [flutter/flutter#84575]
## Steps to Reproduce

1. Create an Android App from scratch in Android Studio.
2. Create a new Flutter module.
3. Bind MainActivity to a new FlutterFragment using [Flutter documentation](https://flutter.dev/docs/development/add-to-app/android/add-flutter-fragment#add-a-flutterfragment-to-an-activity-with-a-new-flutterengine).
4. Launch it and activate Talkback.

**Expected results:** 
- The accessibility outline should be inside the Fragment, over the title "Flutter Demo Home Page"
- Navigating through screen with Talkback, the accessibility outline should point to each Widget in the FlutterFragment.

**Actual results:** 
- The accessibility outline is shown behind the Fragment.
- Navigating through screen with Talkback, the accessibility outline points almost the exact position where the Flutter Widget should be.
- To see this more clearly, add some padding to the Fragment, to make it smaller.

Current Flutter version. But I have reproduced this same issue with previous versions.
```
Flutter (Channel stable, 2.0.6, on macOS 11.1 20C69 darwin-x64, locale
    en-GB)
    • Flutter version 2.0.6
    • Framework revision 1d9032c7e1 (7 weeks ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3
```

This totally works if instead using a FlutterFragment use a FlutterActivity.

**Comment by TahaTesser** [comment]
Hi @dagonco 
Based on your description, it looks like a duplicate of https://github.com/flutter/flutter/issues/75371, which uses `FragmentActivity` 

Can you please confirm if it same? Also it didn't seem to reproduce on Android 10 https://github.com/flutter/flutter/issues/75371#issuecomment-773229735
Thank you

**Comment by dagonco** [comment]
Hi @TahaTesser,

I think the issue is different from the one you linked.
In this case, what happens is that the outline is **behind** the fragment.
It gives the feeling that the highlighted view is shifted to the position it would have if the FlutterFragment were displaying full screen.

I've attached a [video](https://user-images.githubusercontent.com/8106237/122043998-4475c480-cddc-11eb-8d3c-239c0684d931.mp4
) that can expose the issue much better than I can.

PS: I have tested on Android 11 with a Pixel 4XL and it also happens.

**Comment by TahaTesser** [comment]
Hi @dagonco 
Thanks for the video, really helpful. 

I can reproduce the issue when modifying the official Add to app full-screen sample adding FlutterFragmnt


| Preview | 
| --------------- | 
<img src="https://user-images.githubusercontent.com/48603081/122253102-8c781280-ced4-11eb-8c25-856855739b91.gif" height="450" />	|



|  `stable`  | `master` |
| :---: |:---: |
|  ❌  |   ❌  | 

Check `flutter doctor -v` outputs for each channel below

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.2, on Microsoft Windows [Version 10.0.19043.1055], locale en-US)
    • Flutter version 2.2.2 at C:\Users\Taha\Code\flutter_stable
    • Framework revision d79295af24 (5 days ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\openjdk\bin\java.exe
    • Java version OpenJDK Runtime Environment (build 1.8.0-262-b10)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.1)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31402.337
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.57.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.23.0

[✓] Connected device (4 available)
    • IN2011 (mobile)   • c9d8ee0c • android-arm64  • Android 11 (API 30)
    • Windows (desktop) • windows  • windows-x64    • Microsoft Windows [Version 10.0.19043.1055]
    • Chrome (web)      • chrome   • web-javascript • Google Chrome 91.0.4472.101
    • Edge (web)        • edge     • web-javascript • Microsoft Edge 91.0.864.48

! Doctor found issues in 1 category.
```

```console
[✓] Flutter (Channel master, 2.3.0-17.0.pre.375, on Microsoft Windows [Version 10.0.19043.1055], locale en-US)
    • Flutter version 2.3.0-17.0.pre.375 at C:\Users\Taha\Code\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 57a565c706 (14 hours ago), 2021-06-15 18:44:02 -0700
    • Engine revision e0011fa561
    • Dart version 2.14.0 (build 2.14.0-211.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.1)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31402.337
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.2.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.57.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.23.0

[✓] Connected device (5 available)
    • IN2011 (mobile)         • c9d8ee0c • android-arm64   • Android 11 (API 30)
    • Windows (desktop)       • windows  • windows-x64     • Microsoft Windows [Version 10.0.19043.1055]
    • Windows (UWP) (desktop) • winuwp   • windows-uwp-x64 •
    • Chrome (web)            • chrome   • web-javascript  • Google Chrome 91.0.4472.101
    • Edge (web)              • edge     • web-javascript  • Microsoft Edge 91.0.864.48

• No issues found!
```
		
</details>

✅ : No Issue ❌: Issue reproduced

**Comment by chinmaygarde** [comment]
cc @blasten
